src/sqlfluff/rules/L005.py

```diff
@@ -40,10 +41,11 @@ class Rule_L005(BaseRule):
     """
 
     groups = ("all", "core")
+    crawl_behaviour = SegmentSeekerCrawler({"comma"}, provide_raw_stack=True)
 
     def _eval(self, context: RuleContext) -> Optional[LintResult]:
         """Commas should not have whitespace directly before them."""
-        anchor: Optional[RawSegment] = context.raw_segment_pre
+        anchor: Optional[RawSegment] = context.raw_stack[-1]
```

Comment:
It's very unlikely that a file begins with a comma - but I think strictly you're probably right. I'll add a clause.

Reply:
Thanks! I'm not generally paranoid, but for a while, we'd have a bug or two every release where a user would hit one of these unexpected conditions. We added the rules critical errors" test to help catch these things, and it's gotten better since then.